ember-bootstrap/addon/components/bs-dropdown/button.hbs

```diff
@@ -1,21 +1,15 @@
 {{! @glint-nocheck }}
-<button
-  disabled={{this.__disabled}}
+<BsButton
+  {{! cannot pass @block directly because that triggers reserved keyword assertion when compiling the template }}
+  {{! template-lint-disable no-capital-arguments no-args-paths }}
+  @block={{this.args.block}}
+  @onClick={{@onClick}}
   aria-expanded={{if @isOpen "true" "false"}}
-  class="btn dropdown-toggle {{if @active "active"}} {{if this.block "btn-block"}} {{bs-size-class "btn" @size}} {{bs-type-class "btn" @type default=(if (macroCondition (macroGetOwnConfig "isBS3")) "default" "secondary") outline=@outline}}"
-  type="button"
+  class="dropdown-toggle"
   ...attributes
-  {{on "click" this.handleClick}}
   {{on "keydown" @onKeyDown}}
   {{did-insert @registerChildElement "toggle"}}
   {{will-destroy @unregisterChildElement "toggle"}}
 >
-  {{#if this.icon}}<i class={{this.icon}}></i> {{/if}}{{this.text}}{{yield
-    (hash
-      isFulfilled=this.isFulfilled
-      isPending=this.isPending
-      isRejected=this.isRejected
-      isSettled=this.isSettled
-    )
-  }}
-</button>
+  {{yield}}
```

Comment:
This seems like a change in API. The settledness states do not get yield anymore.
Is it possible to add `as |buttonState|` to `<BsButton>` so it can be passed to the `{{yield}}`? (`{{yield buttonState}}`)

Reply:
Only supported usage of `<BsDropdown::Button>` is as the `button` component yielded by `<BsDropdown>`. That component is the toggle for the dropdown. The `@onClick` action can never be async.

The `@onClick` action is hardcoded to be the `toggleDropdown` method of `<BsDropdown>`: https://github.com/ember-bootstrap/ember-bootstrap/blob/9c964025afc85a9d389b70b00a28aa1cfd78ff9a/ember-bootstrap/addon/components/bs-dropdown.hbs#L12-L14

The `toggleDropdown` method does not return a promise: https://github.com/ember-bootstrap/ember-bootstrap/blob/9c964025afc85a9d389b70b00a28aa1cfd78ff9a/ember-bootstrap/addon/components/bs-dropdown.js#L292-L299

It seems that yielding promise state here was done by mistake. Likely an issue due to copying over `<BsButton>`'s template to this template for customization.

Reply:
Ah, I had not realized that the called was hardcoded. In that case it is fine!